state/core_access.go

```diff
@@ -100,16 +101,25 @@ func (ca *CoreAccessor) Start(ctx context.Context) error {
 }
 
 func (ca *CoreAccessor) Stop(context.Context) error {
-	defer ca.cancel()
+	if ca.cancel == nil {
+		log.Warn("core accessor already stopped")
+		return nil
+	}
 	if ca.coreConn == nil {
 		log.Warn("no connection found to close")
 		return nil
 	}
+	defer func() {
+		ca.cancel()
+		ca.cancel = nil
```

Comment:
[non-blocking][recommendation]
When there are multiple steps to something like this, I think it is nice to have a helper function to ensure that, if we need to call cancel in other places, we don't miss a step.
```go
func (ca *CoreAccessor) cancelCTX() {
    ca.cancel()
    ca.cancel = nil
}
```

Reply:
@MSevey done.